velox/dwio/dwrf/common/Common.h

```diff
@@ -271,4 +275,114 @@ class StripeInformation {
   virtual uint64_t getNumberOfRows() const = 0;
 };
 
+static CompressionKind convertCompressionKind(
+    proto::orc::CompressionKind compression) {
+  auto compressionUint = static_cast<uint32_t>(compression);
+  if (compressionUint >= 4 && compressionUint <= 5) {
+    compressionUint = 9 - compressionUint;
+  }
+  return static_cast<CompressionKind>(compressionUint);
+}
+
+class PostScript {
+ public:
+  PostScript(
+      uint64_t footerLength,
+      CompressionKind compression,
+      uint64_t compressionBlockSize,
+      uint32_t writerVersion)
+      : footerLength_{footerLength},
+        compression_{compression},
+        compressionBlockSize_{compressionBlockSize},
+        writerVersion_{static_cast<WriterVersion>(writerVersion)} {}
+
+  explicit PostScript(const proto::PostScript& ps)
+      : PostScript(
+            ps.footerlength(),
+            static_cast<CompressionKind>(ps.compression()),
+            ps.compressionblocksize(),
+            ps.writerversion()) {
+    cacheMode_ = static_cast<StripeCacheMode>(ps.cachemode());
```

Comment:
why not make them also part of the initializater list?

Reply:
My intention was to make use of delegate ctor. Have changed to use initializer list.